lib/src/list_mutations.dart

```diff
@@ -172,14 +176,39 @@ SourceEdit _insertInBlockList(
 
   if (index == list.length) return _appendToBlockList(yamlEdit, list, item);
 
-  final formattedValue = _formatNewBlock(yamlEdit, list, item);
+  var (formattedValue, indent) = _formatNewBlock(yamlEdit, list, item);
 
   final currNode = list.nodes[index];
   final currNodeStart = currNode.span.start.offset;
   final yaml = yamlEdit.toString();
-  final start = yaml.lastIndexOf('\n', currNodeStart) + 1;
 
-  return SourceEdit(start, 0, formattedValue);
+  var currSequenceOffset = yaml.lastIndexOf('-', currNodeStart - 1);
+
+  final (isNested, offset) = _isNestedInBlockList(currSequenceOffset - 1, yaml);
+
+  /// We have to get rid of the left indentation applied by default
+  if (isNested && index == 0) {
+    // Give the previous first element its indent
+    formattedValue = formattedValue.trimLeft() + indent;
+  }
+
+  return SourceEdit(offset, 0, formattedValue);
+}
```

Comment:
So when we're inserting into a block list, we:
 * Find the node that is currently at the index we want to insert at.
 * Find the `-` indicator in front of that node.
 * Determine if we're in a nested block list or not
   * If so, ...?
 * else we....

I think what you're doing is probably correct in both cases, it's just a bit weird.

If I have a nested list:
```
 - - foo
```
and I'm prepending `bar` to the inner list, then I get a diff that looks like:
```
 - - foo
   ^
   Insert: "- bar\n   "
```
Where as if I have:
```
 - foo
```
and I prepend `bar` I get a diff that says:
```
 - foo
^
Insert: " - bar\n"
```

So in the case with no nested lists, we're inserting a full new line.
The `-` indicator that belongs to `foo` moves down.

Where as, if we have nested block lists, we insert into the middle of the existing line.

I guess this makes sense because in `- -` the first `-` belongs to the outer list, and we're not modifying the outer list. We're modifying the inner list.

Just typing this out so I think I understand it, this seems solid.

I'd suggest we either inline `_isNestedInBlockList` or improve the documentation so that it can stand on its own -- otherwise, it's a bit hard to understand what it does when someone reads it.

------------------

I guess technically we could in the second case have done:
```
 - foo
 ^
 Insert: "- bar\n "
```

That would be more consistent with the first case, and we wouldn't need to know if it's a nested list or not. But the diff is much better when we're inserting a full line, and that's also the common case after all, nested lists in block mode is fairly weird. So I totally understand why we need to know if it's inside a nested block list.

Reply:
😅 `yaml` syntax is quirky making everything weird. The existing code works well and assumes people will be civil and use

```yaml
- 
  - value
 ```

instead of 

```yaml
- - value
```

Reply:
oh, I didn't even realize that:
```
- 
  - value
```
was an option... can we add a few tests :rofl:

Reply:
> oh, I didn't even realize that:
> 
> ```
> - 
>   - value
> ```
> 
> was an option... can we add a few tests 🤣

Added these in [5978c20](https://github.com/dart-lang/yaml_edit/pull/84/commits/5978c209143e913e33ff0cf6697a9aa83d2bff4d), I think.

Reply:
Added additional docs